src/V3Param.cpp

```diff
@@ -65,6 +67,295 @@
 
 VL_DEFINE_DEBUG_FUNCTIONS;
 
+//######################################################################
+// Unroll AstGenFor
+
+class UnrollGenVisitor final : public VNVisitor {
```

Comment:
Keep this in V3Unroll.cpp please as they usually need common maintenance.  There's also way too much duplicated code with V3Unroll now.

Reply:
There is nothing in common in the two any more, and you can no longer bolt them together, it's just way more complicated that way and there is no maintenance benefit. How about moving it into V3UnrollGen.cpp if we really must?

> There's also way too much duplicated code with V3Unroll now.

I'm not sure what you mean here, there really isn't.

Reply:
I prefer V3Unroll even if the full class is duplicated, as I still suspect there will be bugs in one side which require also fixing in the other.  I can live with V3UnrollGen though if you prefer.

Reply:
Nothing is duplicated they are different algorithms now :upside_down_face: (and will be even more incompatible once #6280 is finished). I will move this in UnrollGen then, there is nothing shared between the two, keeping them together just makes the reader do mental gymnastics to prove they are independent.